packages/offcanvas/src/offcanvas.ts

```diff
@@ -1,147 +1,108 @@
-        this.options.onHide?.()
-    }
+        this.state = options?.defaultState || this.offcanvasElement.dataset.state || "close";
 
-    private openOffCanvas() {
-        this.options.beforeShow?.()
-        closeAllOpenedOffcanvas(this.offCanvasElement)
-        toggleOffCanvasState(
-            this.offCanvasElement,
-            this.allowBodyScroll,
-            "open")
-        const id = this.offCanvasElement.getAttribute("id") as string
-        const overlayElement = createOverlay(
-            this.backdrop,
-            id
-        )
-        if (overlayElement instanceof HTMLElement) {
-            appendBefore({ newElement: overlayElement, existingElement: this.offCanvasElement })
-            if (!this.staticBackdrop)
-                overlayElement.addEventListener("click", this.closeOffCanvas)
+        if (!this.offcanvasElement.hasAttribute("data-fx-offcanvas")) {
+            this.offcanvasElement.setAttribute("data-fx-offcanvas", "");
         }
-        document.addEventListener("keydown", this.closeWithEsc)
-        this.options.onShow?.()
-    }
 
-    /**
-   * Close the Offcanvas when the "Escape" key is pressed.
-   */
-    private closeWithEsc = (event: KeyboardEvent) => {
-        if (event.key === "Escape") { this.closeOffCanvas() }
-    }
+        const triggerButton = (typeof triggerElement === "string" ? $(triggerElement) : triggerElement) || $(`[data-offcanvas-target='${this.offcanvasElement.dataset.offcanvasId}']`);
 
+        const { showOffcanvas, hideOffcanvas, autoInitOffcanvas, isHidden } = initOffcanvas(offcanvasElement, triggerButton, this.options);
 
-    private initCloseBtns() {
-        for (const closeOffCanvas of this.offCanvasCloseBtns) closeOffCanvas.addEventListener("click", () => this.closeOffCanvas())
-    }
+        // Initialize offcanvas state and animations
+        if (this.state === "open") {
+            this.playOpenAnimation();
+        } else {
+            this.offcanvasElement.style.display = "none";
+        }
 
-    private changeState() {
-        const curState = this.offCanvasElement.getAttribute("data-state")
-        curState === "open" ? this.closeOffCanvas() : this.openOffCanvas()
-    }
+        this.showOffcanvas = this.playOpenAnimation.bind(this);
+        this.hideOffcanvas = this.playCloseAnimation.bind(this);
+        this.isHidden = isHidden;
 
-    private initTriggers() {
-        for (const triggerBtn of this.offCanvasTriggers) triggerBtn.addEventListener("click", () => this.changeState())
-    }
+        this.initTriggers();  // Ensure triggers are initialized
+        this.initCloseBtns();  // Ensure close buttons are initialized
 
-    private setupOffcanvas() {
-        this.initTriggers()
-        this.initCloseBtns()
+        autoInitOffcanvas();
     }
 
-    open() {
-        this.openOffCanvas()
+    // Predefined open animation for offcanvas
```

Comment:
For the Offcanvas, it's the same as the modal, except that currently, there isn't an option to add keyframe animations like there is for the modal.